test/e2e/nodepool_osp_az_test.go

```diff
@@ -105,10 +105,17 @@ func (o OpenStackAZTest) BuildNodePoolManifest(defaultNodepool hyperv1.NodePool)
 	defaultNodepool.Spec.DeepCopyInto(&nodePool.Spec)
 
 	nodePool.Spec.Replicas = &oneReplicas
-	nodePool.Spec.Platform.OpenStack.AvailabilityZone = defaultAvailabilityZone
+	nodePool.Spec.Platform.OpenStack.AvailabilityZone = getAZName()
 	return nodePool, nil
 }
 
+func getAZName() string {
+	if globalOpts.configurableClusterOptions.OpenStackNodeAvailabilityZone != "" {
+		return globalOpts.configurableClusterOptions.OpenStackNodeAvailabilityZone
+	}
+	return defaultAvailabilityZone
```

Comment:
the default AZ "nova" is safe, as it's present in most clouds. Though it's good we can easily override it.